delphin/ace.py

```diff
@@ -252,7 +252,10 @@ def interact(self, datum: str) -> interface.Response:
         Returns:
             :class:`~delphin.interface.Response`
         """
-        validated = self._validate_input(datum)
+        try:
+            validated = self._validate_input(datum)
+        except TypeError:
+            print("TypeError: MRS object not serialized")
```

Comment:
The original issue (#350) was raising a `KeyError` instead of a `TypeError` so this wouldn't catch the correct problem, but also simply printing "TypeError" is not how this should be handled. But this is probably the right place to make a fix.

I suggest doing an explicit type check and raising a `TypeError`. E.g.,

```python
        if not isinstance(datum, str):
            raise TypeError('interact() argument must be a string, '
                            f'not {type(datum).__name__!r}')
        validated = self._validate_input(datum)
```

In this case I think the `TypeError` is the correct error type to raise, instead of a custom exception class. Also, with this `isinstance()` check here, the one in `ACEParser._validate_input()` becomes redundant. It's not wrong to leave it there, but it's slightly less efficient.

Reply:
Thanks for the critiques and recommendations, I'll implement your suggestion soon. Just wondering if you wanted me to also remove the ACEParser._validate_input() method since it becomes redundant.

Reply:
No, don't remove `ACEParser._validate_input()`. It is called by `ACEProcess.interact()`, which is the same for `ACEParser`, `ACETransferer`, and `ACEGenerator`, so removing it would break things. Also there is a second check beyond the `isinstance()` check, and that should stay (to be clear, `ACEParser._validate_input()` should return `datum.strip()`).